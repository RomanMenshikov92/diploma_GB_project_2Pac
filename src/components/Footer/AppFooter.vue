<template lang="pug">
footer.footer
  .container.footer__container
    .footer__logo
      Logo
    .footer__copyright
      Copyright(:copyright="copyrightItem")
</template>

<script lang="ts">
import { computed, defineComponent } from "vue";
import Logo from "@/components/Common/Logo.vue";
import Copyright from "@/components/Footer/Ð¡opyright.vue";
import { useCopyrightStore } from "@/stores/Common/useCopyrightStore";

export default defineComponent({
  name: "AppFooter",
  components: {
    Logo,
    Copyright,
  },
  setup() {
    const copyrightStore = useCopyrightStore();
    const copyrightItem = computed(() => copyrightStore.getCopyrightItem);

    return {
      copyrightItem,
    };
  },
});
</script>

<style lang="sass"></style>
