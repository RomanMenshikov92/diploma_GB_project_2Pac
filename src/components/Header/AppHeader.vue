<template lang="pug">
header.header
  .container.header__container
    NavMenu(:menu="menuList")
</template>

<script lang="ts">
import { defineComponent, computed } from "vue";
import NavMenu from "@/components/Header/NavMenu.vue";
import { useMenuStore } from "@/stores/Common/useMenuStore";

export default defineComponent({
  name: "AppHeader",
  components: {
    NavMenu,
  },
  setup() {
    const menuStore = useMenuStore();
    const menuList = computed(() => menuStore.getMenuItems);

    return {
      menuList,
    };
  },
});
</script>

<style lang="sass"></style>
