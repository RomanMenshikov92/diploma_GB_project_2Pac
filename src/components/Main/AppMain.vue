<template lang="pug">
main.main
  .container.main__container
    router-view
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "AppMain",
});
</script>

<style lang="sass"></style>
