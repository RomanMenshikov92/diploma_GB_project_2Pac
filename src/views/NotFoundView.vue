<template lang="pug">
.not-found
  .not-found__text
    h2.title.title-reset.not-found__title
      NotFoundIcon
    p.paragraph.paragraph-reset.not-found__paragraph Запрашиваемая страница не найдена
    RouterLink.not-found__link(to="/") На главную
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { RouterLink } from "vue-router";
import NotFoundIcon from "@/components/Icon/NotFoundIcon.vue";

export default defineComponent({
  name: "NotFoundView",
  components: {
    RouterLink,
    NotFoundIcon,
  },
  setup() {
    return {};
  },
});
</script>

<style lang="sass" scoped></style>
