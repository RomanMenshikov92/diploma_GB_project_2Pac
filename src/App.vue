<template lang="pug">
.page(:class="{'page-404': is404Page}")
  AppHeader
  AppMain
  AppFooter
</template>

<script lang="ts">
import store from "@/stores/store";
import AppHeader from "@/components/Header/AppHeader.vue";
import AppMain from "@/components/Main/AppMain.vue";
import AppFooter from "@/components/Footer/AppFooter.vue";
import { useRoute } from "vue-router";
import { computed } from "vue";

export default {
  name: "App",
  components: {
    AppHeader,
    AppMain,
    AppFooter,
  },
  setup() {
    const route = useRoute();

    const is404Page = computed(() => {
      if (route.name === null || route.name === undefined) {
        return false;
      }
      return route.name === "not-found";
    });

    return { is404Page };
  },
  store,
};
</script>

<style lang="sass">
@import "@/assets/sass/style.css"
</style>
