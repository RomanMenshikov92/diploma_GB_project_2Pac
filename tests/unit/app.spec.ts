import { shallowMount, mount } from "@vue/test-utils";
import App from "@/App.vue";
import AppHeader from "@/components/Header/AppHeader.vue";
import AppMain from "@/components/Main/AppMain.vue";
import AppFooter from "@/components/Footer/AppFooter.vue";
import { Pinia, createPinia } from "pinia";
import NotFoundView from "@/views/NotFoundView.vue";
import { createRouter, createWebHistory } from "vue-router";

jest.mock("@/components/Header/AppHeader.vue", () => ({
  name: "AppHeader",
  render: jest.fn(),
}));

jest.mock("@/components/Main/AppMain.vue", () => ({
  name: "AppMain",
  render: jest.fn(),
}));

jest.mock("@/components/Footer/AppFooter.vue", () => ({
  name: "AppFooter",
  render: jest.fn(),
}));

jest.mock("@/views/NotFoundView.vue", () => ({
  name: "NotFoundView",
  render: jest.fn(),
}));

describe("App.vue", () => {
  it("renders in App components: AppHeader, AppMain, AppFooter", () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/:pathMatch(.*)",
          name: "not-found",
          component: NotFoundView,
        },
      ],
    });
    const wrapper = shallowMount(App, {
      global: {
        plugins: [router],
      },
    });
    expect(wrapper.findComponent(AppHeader).exists()).toBe(true);
    expect(wrapper.findComponent(AppMain).exists()).toBe(true);
    expect(wrapper.findComponent(AppFooter).exists()).toBe(true);
  });

  // it("renders in App store", () => {
  //   const pinia: Pinia = createPinia();
  //   const wrapper = shallowMount(App, {
  //     global: {
  //       plugins: [pinia],
  //     },
  //   });
  //   expect(wrapper.vm.$pinia).toBeDefined();
  // });

  it("renders correct Pug in app layout", () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/:pathMatch(.*)",
          name: "not-found",
          component: NotFoundView,
        },
      ],
    });
    const pinia: Pinia = createPinia();
    const wrapper = mount(App, {
      global: {
        plugins: [pinia, router],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
