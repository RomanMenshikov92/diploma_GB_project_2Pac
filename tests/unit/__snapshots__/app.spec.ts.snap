// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders correct Pug in app layout 1`] = `
<div class="page">
  <!---->
  <!---->
  <!---->
</div>
`;
